src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,623 @@
+     * @throws IllegalStateException if the file could not be read.
+     */
+    private static Optional<FileText> getFileText(String fileName) {
+        final File file = new File(fileName);
+        FileText result = null;
+
+        if (!file.isDirectory()) {
+            try {
+                result = new FileText(file, StandardCharsets.UTF_8.name());
+            }
+            catch (IOException ex) {
+                throw new IllegalStateException("Cannot read source file: " + fileName, ex);
+            }
+        }
+
+        return Optional.ofNullable(result);
+    }
+
+    /**
+     * Collets all {@link Suppression} instances retrieved from the given {@link FileText}.
+     *
+     * @param fileText {@link FileText} instance.
+     */
+    private void collectSuppressions(FileText fileText) {
+        suppressions.clear();
+
+        for (int lineNo = 0; lineNo < fileText.size(); lineNo++) {
+            final Optional<Suppression> suppression = getSuppression(fileText, lineNo);
+            suppression.ifPresent(suppressions::add);
+        }
+    }
+
+    /**
+     * Tries to extract the suppression from the given line.
+     *
+     * @param fileText {@link FileText} instance.
+     * @param lineNo line number.
+     * @return {@link Optional} of {@link Suppression}.
+     */
+    private Optional<Suppression> getSuppression(FileText fileText, int lineNo) {
+        final String line = fileText.get(lineNo);
+        final Matcher nearbyTextMatcher = nearbyTextPattern.matcher(line);
+
+        Suppression suppression = null;
+        if (nearbyTextMatcher.find()) {
+            final String text = nearbyTextMatcher.group(0);
+            suppression = new Suppression(text, lineNo + 1, this);
+        }
+
+        return Optional.ofNullable(suppression);
+    }
+
+    /**
+     * Finds the nearest {@link Suppression} instance which can suppress
+     * the given {@link AuditEvent}. The nearest suppression is the suppression which scope
+     * is before the line and column of the event.
+     *
+     * @param suppressions collection of {@link Suppression} instances.
+     * @param event {@link AuditEvent} instance.
+     * @return {@link Suppression} instance.
+     */
+    private static Optional<Suppression> getNearestSuppression(Collection<Suppression> suppressions,
+                                                               AuditEvent event) {
+        return suppressions
+                .stream()
+                .filter(suppression -> suppression.isMatch(event))
+                .findFirst();
+    }
+
+    /** The class which represents the suppression. */
+    private static class Suppression {
+
+        /** The first line where warnings may be suppressed. */
+        private final int firstLine;
+
+        /** The last line where warnings may be suppressed. */
+        private final int lastLine;
+
+        /** The regexp which is used to match the event source.*/
+        @XdocsPropertyType(PropertyType.PATTERN)
```

Comment:
`$ mvn clean verify`
```
[ERROR] Failures: 
[ERROR]   XdocsPagesTest.testAllCheckSections:501->validateCheckSection:584->validatePropertySection:725->validatePropertySectionProperties:861->validatePropertySectionPropertyEx:913 config_filters.xml section 'SuppressWithNearbyTextFilter' should have the type for checkPattern
expected: String
but was : Pattern
```

Reply:
Removed it from `Suppression` class.

Reply:
Thanks, the error you were getting was from the main class, but it was the inner-class I was pointing to. The test should not go against the inner-class, otherwise that is a bug with it.